Patrol requires app to use Kotlin 1.8.x, crashes at runtime or buildtime when other version is used

### Steps to reproduce

I've went through the Getting started page and integrated it into my project. 
Then I went to run the test from example:  https://patrol.leancode.co/getting-started#create-a-simple-integration-test

### Actual results

The test fails and logs don't tell much, but the line that is suspicious is `Starting 0 tests on Pixel_3a_API_34(AVD) - 14`.

I've run the same test in the patrol example project and there it works and says `Starting 1 tests on Pixel_3a_API_34(AVD) - 14`. It's hard to compare with that example project because it is very different, for example it uses `patrol(` instead of `patrolTest(` etc, but I retraced all my steps and I have no idea what is wrong.

### Logs

<details>
<summary>Logs</summary>

```
josip.krnjic@Movemedical-V93F4Y7XJQ move-app-flutter %  patrol test -d emulator-5554 --verbose -t integration_test/example_test.dart     

In next major release, patrolTest method will be intended for UI tests.
If you want to use Patrol in your widget tests, use patrol_finders package.

For more information, see https://patrol.leancode.co/patrol-finders-release
Disable this warning by setting the PATROL_FINDERS environment variable.
      
Verbose mode enabled. More logs will be printed.
Received 1 test target(s)
Received test target: /Users/josip.krnjic/IdeaProjects/move-app-flutter/integration_test/example_test.dart
Generated entrypoint /Users/josip.krnjic/IdeaProjects/move-app-flutter/integration_test/test_bundle.dart with 1 bundled test(s)
$ flutter --no-version-check devices --machine
Received 1 device(s) to run on
Received device: emulator-5554
Received 5 --dart-define(s) (0 custom, 5 internal)
Received internal --dart-define: PATROL_WAIT=0
Received internal --dart-define: PATROL_APP_PACKAGE_NAME=com.movemedical.move_app_flutter
Received internal --dart-define: PATROL_APP_BUNDLE_ID=com.example.MyApp
Received internal --dart-define: INTEGRATION_TEST_SHOULD_REPORT_RESULTS_TO_NATIVE=false
Received internal --dart-define: PATROL_TEST_LABEL_ENABLED=true
• Building apk with entrypoint test_bundle.dart...
$ ./gradlew :app:assembleDebug -Ptarget=/Users/josip.krnjic/IdeaProjects/move-app-flutter/integration_test/test_bundle.dart -Pdart-defines=UEFUUk9MX1dBSVQ9MA==,UEFUUk9MX0FQUF9QQUNLQUdFX05BTUU9Y29tLm1vdmVtZWRpY2FsLm1vdmVfYXBwX2ZsdXR0ZXI=,UEFUUk9MX0FQUF9CVU5ETEVfSUQ9Y29tLmV4YW1wbGUuTXlBcHA=,SU5URUdSQVRJT05fVEVTVF9TSE9VTERfUkVQT1JUX1JFU1VMVFNfVE9fTkFUSVZFPWZhbHNl,UEFUUk9MX1RFU1RfTEFCRUxfRU5BQkxFRD10cnVl
        : > Task :app:preBuild UP-TO-DATE
        : > Task :app:preDebugBuild UP-TO-DATE
        : > Task :app:mergeDebugNativeDebugMetadata NO-SOURCE
        : > Task :app:compileFlutterBuildDebug
        : > Task :app:packLibsflutterBuildDebug UP-TO-DATE
        : > Task :audioplayers_android:preBuild UP-TO-DATE
       // deleted lines because comment was too long
        : > Task :app:writeDebugAppMetadata UP-TO-DATE
        : > Task :app:writeDebugSigningConfigVersions UP-TO-DATE
        : > Task :app:compressDebugAssets
        : > Task :app:packageDebug
        : > Task :app:createDebugApkListingFileRedirect UP-TO-DATE
        : > Task :app:assembleDebug
        : 
        : BUILD SUCCESSFUL in 7s
        : 320 actionable tasks: 7 executed, 313 up-to-date
$ ./gradlew :app:assembleDebugAndroidTest -Ptarget=/Users/josip.krnjic/IdeaProjects/move-app-flutter/integration_test/test_bundle.dart -Pdart-defines=UEFUUk9MX1dBSVQ9MA==,UEFUUk9MX0FQUF9QQUNLQUdFX05BTUU9Y29tLm1vdmVtZWRpY2FsLm1vdmVfYXBwX2ZsdXR0ZXI=,UEFUUk9MX0FQUF9CVU5ETEVfSUQ9Y29tLmV4YW1wbGUuTXlBcHA=,SU5URUdSQVRJT05fVEVTVF9TSE9VTERfUkVQT1JUX1JFU1VMVFNfVE9fTkFUSVZFPWZhbHNl,UEFUUk9MX1RFU1RfTEFCRUxfRU5BQkxFRD10cnVl
        : > Task :app:preBuild UP-TO-DATE
        : > Task :app:preDebugBuild UP-TO-DATE
        : > Task :audioplayers_android:preBuild UP-TO-DATE
        : > Task :audioplayers_android:preDebugBuild UP-TO-DATE
        : > Task :audioplayers_android:compileDebugAidl NO-SOURCE
        : > Task :connectivity_plus:preBuild UP-TO-DATE
         // deleted lines because comment was too long
        : > Task :connectivity_plus:bundleLibRuntimeToJarDebug UP-TO-DATE
        : > Task :shared_preferences_android:bundleLibRuntimeToJarDebug UP-TO-DATE
        : > Task :isar_flutter_libs:bundleLibRuntimeToJarDebug UP-TO-DATE
        : > Task :package_info_plus:bundleLibRuntimeToJarDebug UP-TO-DATE
        : > Task :app:validateSigningDebugAndroidTest UP-TO-DATE
        : > Task :app:writeDebugAndroidTestSigningConfigVersions UP-TO-DATE
        : > Task :app:mergeLibDexDebugAndroidTest UP-TO-DATE
        : > Task :app:compileDebugKotlin
        : > Task :app:compileDebugJavaWithJavac UP-TO-DATE
        : > Task :app:bundleDebugClassesToCompileJar UP-TO-DATE
        : > Task :app:compileDebugAndroidTestKotlin NO-SOURCE
        : > Task :app:compileDebugAndroidTestJavaWithJavac UP-TO-DATE
        : > Task :app:mergeDebugAndroidTestJavaResource UP-TO-DATE
        : > Task :app:dexBuilderDebugAndroidTest
        : > Task :app:mergeProjectDexDebugAndroidTest UP-TO-DATE
        : > Task :app:packageDebugAndroidTest UP-TO-DATE
        : > Task :app:createDebugAndroidTestApkListingFileRedirect UP-TO-DATE
        : > Task :app:assembleDebugAndroidTest UP-TO-DATE
        : 
        : BUILD SUCCESSFUL in 1s
        : 322 actionable tasks: 2 executed, 320 up-to-date
✓ Completed building apk with entrypoint test_bundle.dart (9.7s)
Will uninstall apps before running tests
Uninstalling com.movemedical.move_app_flutter from sdk gphone64 arm64
Uninstalling com.movemedical.move_app_flutter.test from sdk gphone64 arm64
• Executing tests of apk with entrypoint test_bundle.dart on emulator-5554...
$ ./gradlew :app:connectedDebugAndroidTest -Ptarget=/Users/josip.krnjic/IdeaProjects/move-app-flutter/integration_test/test_bundle.dart -Pdart-defines=UEFUUk9MX1dBSVQ9MA==,UEFUUk9MX0FQUF9QQUNLQUdFX05BTUU9Y29tLm1vdmVtZWRpY2FsLm1vdmVfYXBwX2ZsdXR0ZXI=,UEFUUk9MX0FQUF9CVU5ETEVfSUQ9Y29tLmV4YW1wbGUuTXlBcHA=,SU5URUdSQVRJT05fVEVTVF9TSE9VTERfUkVQT1JUX1JFU1VMVFNfVE9fTkFUSVZFPWZhbHNl,UEFUUk9MX1RFU1RfTEFCRUxfRU5BQkxFRD10cnVl
        : > Task :app:compileFlutterBuildDebug
        : > Task :app:packLibsflutterBuildDebug UP-TO-DATE
        : > Task :app:preBuild UP-TO-DATE
        : > Task :app:preDebugBuild UP-TO-DATE
        : > Task :audioplayers_android:preBuild UP-TO-DATE
       // deleted lines because comment was too long
        : > Task :app:createDebugAndroidTestApkListingFileRedirect UP-TO-DATE
        : 
        : > Task :app:connectedDebugAndroidTest
        : Starting 0 tests on Pixel_3a_API_34(AVD) - 14
        Instrumentation did not complete:
        Instrumentation code: 0
        Instrumentation result bundle: 
        
        FAILURE: Build failed with an exception.
        
        * What went wrong:
        Execution failed for task ':app:connectedDebugAndroidTest'.
        See the report at: file:///Users/josip.krnjic/IdeaProjects/move-app-flutter/build/app/reports/androidTests/connected/index.html
        
        * Try:
        > Run with --stacktrace option to get the stack trace.
        > Run with --info or --debug option to get more log output.
        > Run with --scan to get full insights.
        
        * Get more help at https://help.gradle.org
        
        BUILD FAILED in 13s
        : 
        : > Task :app:connectedDebugAndroidTest FAILED
        : 345 actionable tasks: 7 executed, 338 up-to-date
✗ Failed to execute tests of apk with entrypoint test_bundle.dart on emulator-5554 (Gradle test execution failed with code 1) (13.5s)
Exception: Gradle test execution failed with code 1
#0      AndroidTestBackend.execute.<anonymous closure> (package:patrol_cli/src/android/android_test_backend.dart:140:9)
<asynchronous suspension>
#1      DisposeScope.run (package:dispose_scope/src/dispose_scope.dart:46:7)
<asynchronous suspension>
#2      AndroidTestBackend.execute (package:patrol_cli/src/android/android_test_backend.dart:106:5)
<asynchronous suspension>
#3      TestCommand._execute (package:patrol_cli/src/commands/test.dart:277:7)
<asynchronous suspension>
#4      TestCommand.run (package:patrol_cli/src/commands/test.dart:169:23)
<asynchronous suspension>
#5      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#6      PatrolCommandRunner.runCommand (package:patrol_cli/src/runner/patrol_command_runner.dart:321:18)
<asynchronous suspension>
#7      PatrolCommandRunner.run (package:patrol_cli/src/runner/patrol_command_runner.dart:265:18)
<asynchronous suspension>
#8      patrolCommandRunner (package:patrol_cli/src/runner/patrol_command_runner.dart:77:20)
<asynchronous suspension>
#9      main (file:///Users/josip.krnjic/.pub-cache/hosted/pub.dev/patrol_cli-2.2.0/bin/main.dart:6:20)
<asynchronous suspension>

See the logs above to learn what happened. Also consider running with --verbose. If the logs still aren't useful, then it's a bug - please report it.
Uninstalling com.movemedical.move_app_flutter from sdk gphone64 arm64
Uninstalling com.movemedical.move_app_flutter.test from sdk gphone64 arm64
josip.krnjic@Movemedical-V93F4Y7XJQ move-app-flutter % 


```

</details>


### Patrol version

patrol: ^2.3.0

### Patrol Doctor output

<details>
<summary>Patrol Doctor output</summary>

```
Patrol CLI version: 2.2.0
Program adb found in /opt/homebrew/bin/adb
Env var $ANDROID_HOME set to /Users/josip.krnjic/Library/Android/sdk
Program xcodebuild found in /usr/bin/xcodebuild
Program ideviceinstaller found in /opt/homebrew/bin/ideviceinstaller

```

</details>


### Flutter Doctor output

<details>
<summary>Flutter Doctor output</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.5, on macOS 13.5.2 22G91 darwin-arm64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] IntelliJ IDEA Community Edition (version 2023.1.5)
[✓] Connected device (4 available)
[✓] Network resources
```

</details>

--- a/packages/patrol/android/build.gradle
+++ b/packages/patrol/android/build.gradle
@@ -63,7 +63,7 @@ android {
         api "androidx.test.espresso:espresso-core:3.5.0"
         api "androidx.test.uiautomator:uiautomator:2.2.0"
 
-        implementation platform("org.http4k:http4k-bom:5.7.4.0")
+        implementation platform("org.http4k:http4k-bom:4.48.0.0")
         implementation "org.http4k:http4k-core"
         implementation "com.squareup.okhttp:okhttp:2.7.5" // See https://github.com/square/okhttp/issues/8031
         implementation "org.http4k:http4k-server-ktorcio"

--- a/packages/patrol/android/build.gradle
+++ b/packages/patrol/android/build.gradle
@@ -63,6 +63,9 @@ android {
         api "androidx.test.espresso:espresso-core:3.5.0"
         api "androidx.test.uiautomator:uiautomator:2.2.0"
 
+        // We need to downgrade http4k-bom to 4.48.0.0 because apps with kotlin version (ext.kotlin_version) lower than 1.8 
+        // end up with compile-time errors. We can find more details in a similar problem:
+        // https://stackoverflow.com/questions/67699823/module-was-compiled-with-an-incompatible-version-of-kotlin-the-binary-version-o
         implementation platform("org.http4k:http4k-bom:4.48.0.0")
         implementation "org.http4k:http4k-core"
         implementation "com.squareup.okhttp:okhttp:2.7.5" // See https://github.com/square/okhttp/issues/8031

--- a/packages/patrol/android/build.gradle
+++ b/packages/patrol/android/build.gradle
@@ -67,7 +67,7 @@ android {
         implementation "org.http4k:http4k-core"
         implementation "com.squareup.okhttp:okhttp:2.7.5" // See https://github.com/square/okhttp/issues/8031
         implementation "org.http4k:http4k-server-ktorcio"
-        implementation 'com.google.code.gson:gson:2.10.1'
+        implementation "com.google.code.gson:gson:2.10.1"
 
         testImplementation "org.jetbrains.kotlin:kotlin-test"
     }

--- a/packages/patrol/android/build.gradle
+++ b/packages/patrol/android/build.gradle
@@ -13,14 +13,12 @@ buildscript {
         classpath "com.android.tools.build:gradle:7.4.2"
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
         classpath "org.jlleitschuh.gradle:ktlint-gradle:11.5.0"
-        classpath "org.jetbrains.kotlin:kotlin-serialization:$kotlin_version"
     }
 }
 
 apply plugin: "com.android.library"
 apply plugin: "kotlin-android"
 apply plugin: "org.jlleitschuh.gradle.ktlint"
-apply plugin: "kotlinx-serialization"
 
 ktlint {
     filter {
@@ -69,7 +67,7 @@ android {
         implementation "org.http4k:http4k-core"
         implementation "com.squareup.okhttp:okhttp:2.7.5" // See https://github.com/square/okhttp/issues/8031
         implementation "org.http4k:http4k-server-ktorcio"
-        implementation "org.jetbrains.kotlinx:kotlinx-serialization-json:1.6.0"
+        implementation 'com.google.code.gson:gson:2.10.1'
 
         testImplementation "org.jetbrains.kotlin:kotlin-test"
     }

--- a/packages/patrol/android/src/main/kotlin/pl/leancode/patrol/PatrolServerFilters.kt
+++ b/packages/patrol/android/src/main/kotlin/pl/leancode/patrol/PatrolServerFilters.kt
@@ -1,14 +1,11 @@
 package pl.leancode.patrol
 
 import androidx.test.uiautomator.UiObjectNotFoundException
-import kotlinx.serialization.json.Json
 import org.http4k.core.Filter
 import org.http4k.core.Response
 import org.http4k.core.Status.Companion.INTERNAL_SERVER_ERROR
 import org.http4k.core.Status.Companion.NOT_FOUND
 import org.http4k.core.Status.Companion.NOT_IMPLEMENTED
-
-private val json = Json { ignoreUnknownKeys = true }
 
 val printer =
     Filter { next ->

--- a/packages/patrol/android/src/main/kotlin/pl/leancode/patrol/contracts/Contracts.kt
+++ b/packages/patrol/android/src/main/kotlin/pl/leancode/patrol/contracts/Contracts.kt
@@ -5,59 +5,48 @@
 
 package pl.leancode.patrol.contracts;
 
-import kotlinx.serialization.Serializable
-
 class Contracts {
-  @Serializable
   enum class GroupEntryType {
     group,
     test,
   }
 
-  @Serializable
   enum class RunDartTestResponseResult {
     success,
     skipped,
     failure,
   }
 
-  @Serializable
   enum class KeyboardBehavior {
     showAndDismiss,
     alternative,
   }
 
-  @Serializable
   enum class HandlePermissionRequestCode {
     whileUsing,
     onlyThisTime,
     denied,
   }
 
-  @Serializable
   enum class SetLocationAccuracyRequestLocationAccuracy {
     coarse,
     fine,
   }
 
-  @Serializable
   data class DartGroupEntry (
     val name: String,
     val type: GroupEntryType,
     val entries: List<DartGroupEntry>
   )
 
-  @Serializable
   data class ListDartTestsResponse (
     val group: DartGroupEntry
   )
 
-  @Serializable
   data class RunDartTestRequest (
     val name: String
   )
 
-  @Serializable
   data class RunDartTestResponse (
     val result: RunDartTestResponseResult,
     val details: String? = null
@@ -67,22 +56,18 @@ class Contracts {
     }
   }
 
-  @Serializable
   data class ConfigureRequest (
     val findTimeoutMillis: Long
   )
 
-  @Serializable
   data class OpenAppRequest (
     val appId: String
   )
 
-  @Serializable
   class OpenQuickSettingsRequest (
 
   )
 
-  @Serializable
   data class Selector (
     val text: String? = null,
     val textStartsWith: String? = null,
@@ -135,13 +120,11 @@ class Contracts {
     }
   }
 
-  @Serializable
   data class GetNativeViewsRequest (
     val selector: Selector,
     val appId: String
   )
 
-  @Serializable
   data class NativeView (
     val className: String? = null,
     val text: String? = null,
@@ -173,18 +156,15 @@ class Contracts {
     }
   }
 
-  @Serializable
   data class GetNativeViewsResponse (
     val nativeViews: List<NativeView>
   )
 
-  @Serializable
   data class TapRequest (
     val selector: Selector,
     val appId: String
   )
 
-  @Serializable
   data class EnterTextRequest (
     val data: String,
     val appId: String,
@@ -200,7 +180,6 @@ class Contracts {
     }
   }
 
-  @Serializable
   data class SwipeRequest (
     val startX: Double,
     val startY: Double,
@@ -209,18 +188,15 @@ class Contracts {
     val steps: Long
   )
 
-  @Serializable
   data class WaitUntilVisibleRequest (
     val selector: Selector,
     val appId: String
   )
 
-  @Serializable
   data class DarkModeRequest (
     val appId: String
   )
 
-  @Serializable
   data class Notification (
     val appName: String? = null,
     val title: String,
@@ -235,17 +211,14 @@ class Contracts {
     }
   }
 
-  @Serializable
   data class GetNotificationsResponse (
     val notifications: List<Notification>
   )
 
-  @Serializable
   class GetNotificationsRequest (
 
   )
 
-  @Serializable
   data class TapOnNotificationRequest (
     val index: Long? = null,
     val selector: Selector? = null
@@ -258,22 +231,18 @@ class Contracts {
     }
   }
 
-  @Serializable
   data class PermissionDialogVisibleResponse (
     val visible: Boolean
   )
 
-  @Serializable
   data class PermissionDialogVisibleRequest (
     val timeoutMillis: Long
   )
 
-  @Serializable
   data class HandlePermissionRequest (
     val code: HandlePermissionRequestCode
   )
 
-  @Serializable
   data class SetLocationAccuracyRequest (
     val locationAccuracy: SetLocationAccuracyRequestLocationAccuracy
   )

--- a/packages/patrol/android/src/main/kotlin/pl/leancode/patrol/contracts/NativeAutomatorServer.kt
+++ b/packages/patrol/android/src/main/kotlin/pl/leancode/patrol/contracts/NativeAutomatorServer.kt
@@ -5,13 +5,12 @@
 
 package pl.leancode.patrol.contracts;
 
+import com.google.gson.Gson
 import org.http4k.core.Response
 import org.http4k.core.Method.POST
 import org.http4k.routing.bind
 import org.http4k.core.Status.Companion.OK
 import org.http4k.routing.routes
-import kotlinx.serialization.encodeToString
-import kotlinx.serialization.json.Json
 
 abstract class NativeAutomatorServer {
     abstract fun initialize()
@@ -55,7 +54,7 @@ abstract class NativeAutomatorServer {
         Response(OK)
       },
       "configure" bind POST to {
-        val body = json.decodeFromString<Contracts.ConfigureRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.ConfigureRequest::class.java)
         configure(body)
         Response(OK)
       },
@@ -76,42 +75,42 @@ abstract class NativeAutomatorServer {
         Response(OK)
       },
       "openApp" bind POST to {
-        val body = json.decodeFromString<Contracts.OpenAppRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.OpenAppRequest::class.java)
         openApp(body)
         Response(OK)
       },
       "openQuickSettings" bind POST to {
-        val body = json.decodeFromString<Contracts.OpenQuickSettingsRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.OpenQuickSettingsRequest::class.java)
         openQuickSettings(body)
         Response(OK)
       },
       "getNativeViews" bind POST to {
-        val body = json.decodeFromString<Contracts.GetNativeViewsRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.GetNativeViewsRequest::class.java)
         val response = getNativeViews(body)
-        Response(OK).body(json.encodeToString(response))
+        Response(OK).body(json.toJson(response))
       },
       "tap" bind POST to {
-        val body = json.decodeFromString<Contracts.TapRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.TapRequest::class.java)
         tap(body)
         Response(OK)
       },
       "doubleTap" bind POST to {
-        val body = json.decodeFromString<Contracts.TapRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.TapRequest::class.java)
         doubleTap(body)
         Response(OK)
       },
       "enterText" bind POST to {
-        val body = json.decodeFromString<Contracts.EnterTextRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.EnterTextRequest::class.java)
         enterText(body)
         Response(OK)
       },
       "swipe" bind POST to {
-        val body = json.decodeFromString<Contracts.SwipeRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.SwipeRequest::class.java)
         swipe(body)
         Response(OK)
       },
       "waitUntilVisible" bind POST to {
-        val body = json.decodeFromString<Contracts.WaitUntilVisibleRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.WaitUntilVisibleRequest::class.java)
         waitUntilVisible(body)
         Response(OK)
       },
@@ -148,12 +147,12 @@ abstract class NativeAutomatorServer {
         Response(OK)
       },
       "enableDarkMode" bind POST to {
-        val body = json.decodeFromString<Contracts.DarkModeRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.DarkModeRequest::class.java)
         enableDarkMode(body)
         Response(OK)
       },
       "disableDarkMode" bind POST to {
-        val body = json.decodeFromString<Contracts.DarkModeRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.DarkModeRequest::class.java)
         disableDarkMode(body)
         Response(OK)
       },
@@ -170,27 +169,27 @@ abstract class NativeAutomatorServer {
         Response(OK)
       },
       "getNotifications" bind POST to {
-        val body = json.decodeFromString<Contracts.GetNotificationsRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.GetNotificationsRequest::class.java)
         val response = getNotifications(body)
-        Response(OK).body(json.encodeToString(response))
+        Response(OK).body(json.toJson(response))
       },
       "tapOnNotification" bind POST to {
-        val body = json.decodeFromString<Contracts.TapOnNotificationRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.TapOnNotificationRequest::class.java)
         tapOnNotification(body)
         Response(OK)
       },
       "isPermissionDialogVisible" bind POST to {
-        val body = json.decodeFromString<Contracts.PermissionDialogVisibleRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.PermissionDialogVisibleRequest::class.java)
         val response = isPermissionDialogVisible(body)
-        Response(OK).body(json.encodeToString(response))
+        Response(OK).body(json.toJson(response))
       },
       "handlePermissionDialog" bind POST to {
-        val body = json.decodeFromString<Contracts.HandlePermissionRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.HandlePermissionRequest::class.java)
         handlePermissionDialog(body)
         Response(OK)
       },
       "setLocationAccuracy" bind POST to {
-        val body = json.decodeFromString<Contracts.SetLocationAccuracyRequest>(it.bodyString())
+        val body = json.fromJson(it.bodyString(), Contracts.SetLocationAccuracyRequest::class.java)
         setLocationAccuracy(body)
         Response(OK)
       },
@@ -204,6 +203,6 @@ abstract class NativeAutomatorServer {
       }
     )
 
-    private val json = Json { ignoreUnknownKeys = true }
+    private val json = Gson()
 }
 

--- a/packages/patrol/android/src/main/kotlin/pl/leancode/patrol/contracts/PatrolAppServiceClient.kt
+++ b/packages/patrol/android/src/main/kotlin/pl/leancode/patrol/contracts/PatrolAppServiceClient.kt
@@ -5,24 +5,23 @@
 
 package pl.leancode.patrol.contracts;
 
+import com.google.gson.Gson
 import com.squareup.okhttp.MediaType
 import com.squareup.okhttp.OkHttpClient
 import com.squareup.okhttp.Request
 import com.squareup.okhttp.RequestBody
-import kotlinx.serialization.encodeToString
-import kotlinx.serialization.json.Json
 import java.util.concurrent.TimeUnit
 
 class PatrolAppServiceClient(address: String, port: Int, private val timeout: Long, private val timeUnit: TimeUnit) {
 
     fun listDartTests(): Contracts.ListDartTestsResponse {
         val response = performRequest("listDartTests")
-        return json.decodeFromString(response)
+        return json.fromJson(response, Contracts.ListDartTestsResponse::class.java)
     }
 
     fun runDartTest(request: Contracts.RunDartTestRequest): Contracts.RunDartTestResponse {
-        val response = performRequest("runDartTest", json.encodeToString(request))
-        return json.decodeFromString(response)
+        val response = performRequest("runDartTest", json.toJson(request))
+        return json.fromJson(response, Contracts.RunDartTestResponse::class.java)
     }
 
     private fun performRequest(path: String, requestBody: String? = null): String {
@@ -53,7 +52,7 @@ class PatrolAppServiceClient(address: String, port: Int, private val timeout: Lo
 
     val serverUrl = "http://$address:$port/"
 
-    private val json = Json { ignoreUnknownKeys = true }
+    private val json = Gson()
 
     private val jsonMediaType = MediaType.parse("application/json; charset=utf-8")
 }

--- a/packages/patrol_gen/lib/src/generators/android/android_contracts_generator.dart
+++ b/packages/patrol_gen/lib/src/generators/android/android_contracts_generator.dart
@@ -33,8 +33,6 @@ class AndroidContractsGenerator {
 
 package ${config.package};
 
-import kotlinx.serialization.Serializable
-
 ''';
   }
 
@@ -59,7 +57,6 @@ $optionalFieldUtils
     }
 
     return '''
-  @Serializable
   ${dataKeyword}class ${message.name} (
 $fields
   )$optionalFieldUtils
@@ -77,7 +74,6 @@ $fields
     final cases = enumDefinition.fields.map((e) => '    $e,').join('\n');
 
     return '''
-  @Serializable
   enum class ${enumDefinition.name} {
 $cases
   }

--- a/packages/patrol_gen/lib/src/generators/android/android_http4k_client_generator.dart
+++ b/packages/patrol_gen/lib/src/generators/android/android_http4k_client_generator.dart
@@ -25,12 +25,11 @@ class AndroidHttp4kClientGenerator {
 
 package ${config.package};
 
+import com.google.gson.Gson
 import com.squareup.okhttp.MediaType
 import com.squareup.okhttp.OkHttpClient
 import com.squareup.okhttp.Request
 import com.squareup.okhttp.RequestBody
-import kotlinx.serialization.encodeToString
-import kotlinx.serialization.json.Json
 import java.util.concurrent.TimeUnit
 
 ''';
@@ -77,7 +76,7 @@ $endpoints
 
     val serverUrl = $url
 
-    private val json = Json { ignoreUnknownKeys = true }
+    private val json = Gson()
 
     private val jsonMediaType = MediaType.parse("application/json; charset=utf-8")
 }''';
@@ -92,12 +91,12 @@ $endpoints
         : '';
 
     final serializeParameter =
-        endpoint.request != null ? ', json.encodeToString(request)' : '';
+        endpoint.request != null ? ', json.toJson(request)' : '';
 
     final body = endpoint.response != null
         ? '''
         val response = performRequest("${endpoint.name}"$serializeParameter)
-        return json.decodeFromString(response)'''
+        return json.fromJson(response, Contracts.${endpoint.response!.name}::class.java)'''
         : '''
         return performRequest("${endpoint.name}"$serializeParameter)''';
 

--- a/packages/patrol_gen/lib/src/generators/android/android_http4k_server_generator.dart
+++ b/packages/patrol_gen/lib/src/generators/android/android_http4k_server_generator.dart
@@ -23,13 +23,12 @@ class AndroidHttp4kServerGenerator {
 
 package ${config.package};
 
+import com.google.gson.Gson
 import org.http4k.core.Response
 import org.http4k.core.Method.POST
 import org.http4k.routing.bind
 import org.http4k.core.Status.Companion.OK
 import org.http4k.routing.routes
-import kotlinx.serialization.encodeToString
-import kotlinx.serialization.json.Json
 
 ''';
   }
@@ -46,7 +45,7 @@ $handlers
 $routes
     )
 
-    private val json = Json { ignoreUnknownKeys = true }
+    private val json = Gson()
 }
 ''';
   }
@@ -56,12 +55,12 @@ $routes
       final requestDeserialization = e.request != null
           ? '''
 
-        val body = json.decodeFromString<Contracts.${e.request!.name}>(it.bodyString())'''
+        val body = json.fromJson(it.bodyString(), Contracts.${e.request!.name}::class.java)'''
           : '';
       final requestArg = e.request != null ? 'body' : '';
 
       final responseSerialization =
-          e.response != null ? '.body(json.encodeToString(response))' : '';
+          e.response != null ? '.body(json.toJson(response))' : '';
 
       final responseVariable = e.response != null ? 'val response = ' : '';
 

